atlasdb-jepsen-tests/src/main/java/com/palantir/atlasdb/jepsen/lock/RefreshCorrectnessChecker.java

```diff
@@ -0,0 +1,147 @@
+        private final List<Event> errors = new ArrayList<>();
+        private final Map<Integer, String> resourceName = new HashMap<>();
+
+        @Override
+        public void visit(InfoEvent event) {
+        }
+
+        @Override
+        public void visit(InvokeEvent event) {
+            Integer process = event.process();
+            String lockName = event.value();
+            resourceName.put(process, lockName);
+            pendingForProcessAndLock.put(new Pair(process, lockName), event);
+
+            if (!allLockNames.contains(lockName)) {
+                allLockNames.add(lockName);
+                locksHeld.put(lockName, TreeRangeSet.create());
+            }
+        }
+
+        @Override
+        public void visit(OkEvent event) {
+            Integer process = event.process();
+            String lockName = resourceName.get(process);
+            Pair processLock = new Pair(process, lockName);
+            InvokeEvent invokeEvent = pendingForProcessAndLock.get(processLock);
+
+            switch (event.function()) {
+                /**
+                 * Successful LOCK:
+                 * Remember the new value for the most recent successful lock
+                 */
+                case RequestType.LOCK:
+                    if (event.isSuccessful()) {
+                        lastHeldLock.put(processLock, event);
+                    }
+                    break;
+                /**
+                 * Successful REFRESH/UNLOCK:
+                 * Add the new interval [a, b) to the set of known locks, where
+                 * a is the OkEvent.time() of the last successful lock or the InvokeEvent.time() of refresh, and
+                 * b is the InvokeEvent.time() of the current request, if and only if b > a.
+                 * Also verify that the whole interval was free. Unlock can be treated as refresh, as the correctness
+                 * of their mutual interaction is verified by IsolatedProcessCorrectnessChecker
+                 */
+                case RequestType.REFRESH:
+                case RequestType.UNLOCK:
+                    if (event.isSuccessful() && lastHeldLock.containsKey(processLock)) {
+                        long lastLockTime = lastHeldLock.get(processLock).time();
+                        if (lastLockTime < invokeEvent.time()) {
+                            Range<Long> newRange = Range.closedOpen(lastLockTime, invokeEvent.time());
+                            if (!locksHeld.get(lockName).complement().encloses(newRange)) {
+                                log.error("A {} request for lock {} by process {} invoked at time {} was granted at "
+                                        + "time {}, but another process was granted the lock between {} and {} "
+                                        + "(last known time the lock was held by {})",
+                                        invokeEvent.function(), invokeEvent.value(), invokeEvent.process(),
```

Comment:
I think the first 4 substitutions here should be `event` (dealing with the current event that was successful when it shouldn't have been), not `invokeEvent`.